runtime/binding-grpc-kafka/src/main/java/io/aklivity/zilla/runtime/binding/grpc/kafka/internal/stream/GrpcKafkaProxyFactory.java

```diff
@@ -242,13 +242,26 @@ public MessageConsumer newStream(
                 final GrpcKafkaWithProduceResult result =
                     route.with.resolveProduce(authorization, grpcBeginEx);
 
-                newStream = new GrpcProduceProxy(
-                    grpc,
-                    originId,
-                    routedId,
-                    initialId,
-                    resolvedId,
-                    result)::onGrpcMessage;
+                if (result.replyTo() != null && result.replyTo().value() != null)
```

Comment:
Perhaps add a convenience method `result.hasReplyTo()` with this implementation, so we can avoid repeating the logic in each place that needs to be conditional about presence of replyTo.